tensorpipe/common/cuda_host_allocator.cc

```diff
@@ -51,70 +55,62 @@ void CudaHostAllocator::close() {
     return;
   }
   closed_ = true;
-  cv_.notify_all();
+  processAllocations(std::move(lock));
 }
 
 void CudaHostAllocator::join() {
   close();
   if (!joined_.exchange(true)) {
-    thread_.join();
+    std::unique_lock<std::mutex> lock(mutex_);
+    cv_.wait(lock, [this]() {
+      return pendingAllocations_.empty() && (allocatedChunks_ == 0);
+    });
```

Comment:
What is the point of this join method? It waits without doing anything

Reply:
The point was to ensure the Allocator remains alive until all chunks have been reclaimed, but we can rely on the Context outliving the channels for this. Removing.